### ISSUE TITLE
Tests fail if used with MariaDB

### ISSUE THREAD

**Issue by leptoceratops** [biopython/biopython#1796]
### Setup

I am reporting a problem with Biopython version 1.72, Python version 2.7, and operating
system cygwin as follows:

```
# Edited setup_BioSQL.py as follows
# -- MySQL
DBDRIVER = 'mysql.connector'
DBTYPE = 'mysql'

# Constants for the database driver
DBHOST = 'localhost'
DBUSER = 'root'
DBPASSWD = ''
TESTDB = 'bioseqdb'
```

and I ran:

```
python run_tests.py test_BioSQL test_BioSQL_SeqIO
```

### Expected behaviour

The expected behaviour is to PASS the test

### Actual behaviour

```
test_BioSQL ... FAIL
test_BioSQL_SeqIO ... ERROR
======================================================================
ERROR: test_record_loading (test_BioSQL.InDepthLoadTest)
Make sure all records are correctly loaded.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 346, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_seq_feature (test_BioSQL.InDepthLoadTest)
Indepth check that SeqFeatures are transmitted through the db.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 346, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_load_database (test_BioSQL.LoaderTest)
Load SeqRecord objects into a BioSQL database.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 292, in setUp
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_get_db_items (test_BioSQL.ReadTest)
Get a list of all items in the database.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 115, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_lookup_items (test_BioSQL.ReadTest)
Test retrieval of items using various ids.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 115, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_addition (test_BioSQL.SeqInterfaceTest)
Check can add DBSeq objects together.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 166, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_convert (test_BioSQL.SeqInterfaceTest)
Check can turn a DBSeq object into a Seq or MutableSeq.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 166, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_seq (test_BioSQL.SeqInterfaceTest)
Make sure Seqs from BioSQL implement the right interface.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 166, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_seq_features (test_BioSQL.SeqInterfaceTest)
Check SeqFeatures of a sequence.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 166, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_seq_record (test_BioSQL.SeqInterfaceTest)
Make sure SeqRecords from BioSQL implement the right interface.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 166, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_seq_slicing (test_BioSQL.SeqInterfaceTest)
Check that slices of sequences are retrieved properly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kl344/Tests/test_BioSQL.py", line 166, in setUp
    load_database(gb_handle)
  File "/home/kl344/Tests/test_BioSQL.py", line 90, in load_database
    create_database()
  File "/home/kl344/Tests/test_BioSQL.py", line 80, in create_database
    server.load_database_sql(SQL_FILE)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 311, in load_database_sql
    self.adaptor.cursor.execute(sql_line)
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near 'TYPE=INNODB' at line 1

======================================================================
ERROR: test_BioSQL_SeqIO
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_tests.py", line 244, in runTest
    suite = unittest.TestLoader().loadTestsFromName(name)
  File "/usr/lib/python2.7/unittest/loader.py", line 91, in loadTestsFromName
    module = __import__('.'.join(parts_copy))
  File "/home/kl344/Tests/test_BioSQL_SeqIO.py", line 420, in <module>
    if db_name in server.keys() :
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 202, in keys
    return self.adaptor.list_biodatabase_names()
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 536, in list_biodatabase_names
    "SELECT name FROM biodatabase")
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 641, in execute_and_fetch_col0
    out = super(MysqlConnectorAdaptor, self).execute_and_fetch_col0(sql, args)
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 614, in execute_and_fetch_col0
    self.execute(sql, args or ())
  File "/usr/lib/python2.7/site-packages/BioSQL/BioSeqDatabase.py", line 580, in execute
    self.dbutils.execute(self.cursor, sql, args)
  File "/usr/lib/python2.7/site-packages/BioSQL/DBUtils.py", line 46, in execute
    cursor.execute(sql, args or ())
  File "/usr/lib/python2.7/site-packages/mysql/connector/cursor.py", line 559, in execute
    self._handle_result(self._connection.cmd_query(stmt))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 494, in cmd_query
    result = self._handle_result(self._send_cmd(ServerCmd.QUERY, query))
  File "/usr/lib/python2.7/site-packages/mysql/connector/connection.py", line 396, in _handle_result
    raise errors.get_exception(packet)
ProgrammingError: 1146 (42S02): Table 'bioseqdb.biodatabase' doesn't exist
----------------------------------------------------------------------
Ran 2 tests in 11.690 seconds

FAILED (failures = 2)
```

### Steps to reproduce

```
# Edited setup_BioSQL.py as follows
# -- MySQL
DBDRIVER = 'mysql.connector'
DBTYPE = 'mysql'

# Constants for the database driver
DBHOST = 'localhost'
DBUSER = 'root'
DBPASSWD = ''
TESTDB = 'bioseqdb'
```

and I ran:

```
python run_tests.py test_BioSQL test_BioSQL_SeqIO
```

**Comment by peterjc** [comment]
Could you run ``python`` and report the output of these commands as requested in the issue template:

```
import sys; print(sys.version)
import platform; print(platform.python_implementation()); print(platform.platform())
import Bio; print(Bio.__version__)
```

(This should help rule out there being more than one copy of Biopython installed, and that the versions you think you are using are what is being used)

**Comment by leptoceratops** [comment]
>>> import sys; print(sys.version)
2.7.14 (default, Oct 31 2017, 21:12:13)
[GCC 6.4.0]
 >>> import platform; print(platform.python_implementation()); 
print(platform.platform())
import Bio; print(Bio.__version__)CPython
CYGWIN_NT-10.0-2.11.1-0.329-5-3-x86_64-64bit
 >>> import Bio; print(Bio.__version__)
1.72

**Comment by peterjc** [comment]
Also, to confirm the SQL syntax being run here, what does this say:

```
$ grep INNODB /home/kl344/Tests/biosqldb-mysql.sql
```

(I'm struggling to see where ``TYPE=INNODB`` is coming from, as the SQL script included with Biopython 1.72 should be using ``ENGINE=INNODB``)

**Comment by peterjc** [comment]
Resolved by email - Biopython 1.72 was installed with pip, but the tests were from an older fork of the main repository rather than https://github.com/biopython/biopython/ (and predated 42c95a769971b55628afed15ab414259612e6ae6 and thus was missing the TYPE/ENGINE change).
